vnext/Microsoft.ReactNative/Base/CoreNativeModules.cpp

```diff
@@ -46,65 +46,68 @@ bool HasPackageIdentity() noexcept {
     Mso::CntPtr<AppearanceChangeListener> &&appearanceListener,
     const std::shared_ptr<IReactInstance> &uwpInstance) noexcept {
-  // Modules
   std::vector<facebook::react::NativeModuleDescription> modules;
 
   modules.emplace_back(
       "UIManager",
       [uiManager, uiMessageQueue]() {
         return facebook::react::createUIManagerModule(std::shared_ptr(uiManager), std::shared_ptr(uiMessageQueue));
       },
-      messageQueue);
+      batchingUIMessageQueue);
 
   modules.emplace_back(
       react::uwp::WebSocketModule::Name,
       []() { return std::make_unique<react::uwp::WebSocketModule>(); },
-      MakeSerialQueueThread());
+      jsMessageQueue);
 
-  modules.emplace_back(
-      NetworkingModule::Name, []() { return std::make_unique<NetworkingModule>(); }, MakeSerialQueueThread());
+  modules.emplace_back(NetworkingModule::Name, []() { return std::make_unique<NetworkingModule>(); }, jsMessageQueue);
 
   modules.emplace_back(
-      "Timing", [messageQueue]() { return facebook::react::CreateTimingModule(messageQueue); }, messageQueue);
+      "Timing",
+      [batchingUIMessageQueue]() { return facebook::react::CreateTimingModule(batchingUIMessageQueue); },
+      batchingUIMessageQueue);
 
   modules.emplace_back(
-      LinkingManagerModule::name, []() { return std::make_unique<LinkingManagerModule>(); }, messageQueue);
+      LinkingManagerModule::name, []() { return std::make_unique<LinkingManagerModule>(); }, batchingUIMessageQueue);
 
   modules.emplace_back(
       ImageViewManagerModule::name,
-      [messageQueue]() { return std::make_unique<ImageViewManagerModule>(); },
-      messageQueue);
+      []() { return std::make_unique<ImageViewManagerModule>(); },
+      batchingUIMessageQueue);
 
   modules.emplace_back(
       LocationObserverModule::name,
-      [messageQueue]() { return std::make_unique<LocationObserverModule>(messageQueue); },
-      MakeSerialQueueThread()); // TODO: figure out threading
+      [batchingUIMessageQueue]() { return std::make_unique<LocationObserverModule>(batchingUIMessageQueue); },
+      jsMessageQueue); // TODO: figure out threading
 
   modules.emplace_back(
       react::uwp::AppThemeModule::Name,
       [appTheme = std::move(appTheme)]() mutable {
         return std::make_unique<react::uwp::AppThemeModule>(std::move(appTheme));
       },
-      messageQueue);
+      batchingUIMessageQueue);
 
   modules.emplace_back(
       NativeAnimatedModule::name,
       [wpUwpInstance = std::weak_ptr(uwpInstance)]() mutable {
         return std::make_unique<NativeAnimatedModule>(std::move(wpUwpInstance));
       },
-      messageQueue);
+      batchingUIMessageQueue);
 
   modules.emplace_back(
       AppearanceModule::Name,
       [appearanceListener = std::move(appearanceListener)]() mutable {
         return std::make_unique<AppearanceModule>(std::move(appearanceListener));
       },
-      messageQueue);
+      jsMessageQueue);
```

Comment:
Threading here changed. The module was designed to work off-UI-thread, so we shouldn't affine it to the UI thread.